### PR TITLE
MinecraftClient.openScreen -> setScreen

### LINKED ISSUE #1597: MinecraftClient.openScreen -> setScreen
It first closes the current screen. `openScreen` makes it sound as if the previous screen stays open, and `setScreen(null)` makes more sense than `openScreen(null)`.

### PR DESCRIPTION
**Pull request by Juuxel**
Closes #1597. The issue has the main reasoning, but I'd also mention that an error string within the method calls it `setScreen`.

This is also a semi-impactful rename, though less than #2511 so I don't think it needs to be delayed to after a release.

Also fixes a tiny javadoc error in `Item.Settings`.

### CODE REVIEW

File: mappings/net/minecraft/client/MinecraftClient.mapping [hunk 1]
```diff
@@ -200,12 +200,15 @@ CLASS net/minecraft/class_310 net/minecraft/client/MinecraftClient
...
-	METHOD method_1507 openScreen (Lnet/minecraft/class_437;)V
-		COMMENT Opens a new screen, changing the current screen if needed.
+	METHOD method_1507 setScreen (Lnet/minecraft/class_437;)V
+		COMMENT Sets the current screen to a new screen.
 		COMMENT
 		COMMENT <p>If the screen being opened is {@code null} and the client is not in game, the title screen will be opened.
-		COMMENT If the currently opened screen is {@code null} and player is dead then the death screen will be opened.
-		COMMENT Otherwise the currently open screen will be closed.
+		COMMENT If the currently opened screen is {@code null} and player is dead, then the death screen will be opened.
+		COMMENT Otherwise, the currently open screen will be closed.
```
Comment (liach) [inline mappings/net/minecraft/client/MinecraftClient.mapping]: I suggest replace `currently` with `previous` to make it more clear (not confuse current with the passed parameter)

File: mappings/net/minecraft/client/MinecraftClient.mapping [hunk 2]
```diff
@@ -204,8 +204,8 @@ CLASS net/minecraft/class_310 net/minecraft/client/MinecraftClient
 		COMMENT Sets the current screen to a new screen.
 		COMMENT
 		COMMENT <p>If the screen being opened is {@code null} and the client is not in game, the title screen will be opened.
-		COMMENT If the currently opened screen is {@code null} and player is dead, then the death screen will be opened.
-		COMMENT Otherwise, the currently open screen will be closed.
+		COMMENT If the previous screen is {@code null} and the player is dead, then the death screen will be opened.
```
Comment (liach) [inline mappings/net/minecraft/client/MinecraftClient.mapping]: So the death screen would override the provided parameter screen when the player is dead, I assume?

File: mappings/net/minecraft/client/MinecraftClient.mapping [hunk 3]
```diff
@@ -200,12 +200,15 @@ CLASS net/minecraft/class_310 net/minecraft/client/MinecraftClient
...
+	METHOD method_1507 setScreen (Lnet/minecraft/class_437;)V
+		COMMENT Sets the current screen to a new screen.
 		COMMENT
 		COMMENT <p>If the screen being opened is {@code null} and the client is not in game, the title screen will be opened.
-		COMMENT If the currently opened screen is {@code null} and player is dead then the death screen will be opened.
-		COMMENT Otherwise the currently open screen will be closed.
+		COMMENT If the previous screen is {@code null} and the player is dead, then the death screen will be opened.
+		COMMENT Otherwise, the previous screen will be closed.
+		COMMENT
+		COMMENT <p>If there is an open screen when the current screen is changed, {@link Screen#removed()} will
```
Comment (liach) [inline mappings/net/minecraft/client/MinecraftClient.mapping]: Consider merging this and the previous sentence, both of which talk about screen removal.

File: mappings/net/minecraft/client/MinecraftClient.mapping (hunk 2 above)
Comment (Juuxel) [reply mappings/net/minecraft/client/MinecraftClient.mapping]: No, see the line above - this only applies when the new screen is null.

Comment (liach) [reply mappings/net/minecraft/client/MinecraftClient.mapping]: Hmm, so a table format would be better:
```
<p>If the new screen is {@code null}:
<ul>
<li>if the client is not in game, the title screen will be opened;</li>
<li>if the previous screen is {@code null} and the player is dead, then the death screen will be opened.</li>
</ul>
```

**Comment by Juuxel** [comment]
@liach I changed the docs a bit again, and also fixed a factual error about the death screen.

Rendered:
rendered JD

(The double `@Nullable` happens elsewhere too, not sure what causes it)

**Review by liach — APPROVED** [review]

**Comment by liach** [comment]
> (The double @Nullable happens elsewhere too, not sure what causes it)

Fyi this is because the jetbrains nullable is both type-use and parameter capable. Hence, when mappingpoet generates `@Nullable Type value`, javadoc would interpret that nullable to be used for both cases, and hence it's rendered twice.

**Review by FoundationGames — APPROVED** [review]

**Review by enbrain — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]
I like this change, as this is qutie impactful I think its best we hold off on merging it untill we are also happy to merge: https://github.com/FabricMC/yarn/pull/2511

**Review by YanisBft — APPROVED** [review]

**Pull request merged**